music21/osmd/osmd.py

```diff
@@ -0,0 +1,222 @@
+del importlib
+
+
+def getExtendedModules():
+    # pylint: disable=import-outside-toplevel
+    if hasInstalledIPython:
+        from IPython.core.display import display, HTML, Javascript
+    else:
+        def display():
+            raise OpenSheetMusicDisplayException(
+                'OpenSheetMusicDisplay requires IPython to be installed')
+
+        HTML = Javascript = display
+    return display, HTML, Javascript
+
+
+class ConverterOpenSheetMusicDisplay(SubConverter):
+    '''
+    Converter for displaying sheet music inline
+    in an IPython notebook.
+    Uses: https://opensheetmusicdisplay.org/ for rendering
+    '''
+    registerFormats = ('osmd',)
+    registerShowFormats = ('osmd',)
+    # when updating the script tag osmd will only reload in a notebook if you: Kernel -> restart and
+    # clear output, then save the notebook and refresh the page.
+    # Otherwise the script will stay on the page and not reload.
+    script_url = ("https://github.com/opensheetmusicdisplay"
+                  + "/opensheetmusicdisplay/releases/download/0.9.2/opensheetmusicdisplay.min.js")
+    osmd_file = os.path.join(getSourceFilePath(), 'osmd', 'opensheetmusicdisplay.0.9.2.min.js')
+
+    def __init__(self):
+        super().__init__()
+        self.display, self.HTML, self.Javascript = getExtendedModules()
+
+    def show(self, obj, fmt,
+             fixPartName=True, offline=False, divId=None,
+             **keywords):
+        r'''
+        Displays the score object in a notebook using the OpenSheetMusicDisplay.js library.
+
+        >>> import music21
+        >>> s = music21.converter.parse("tinyNotation: 3/4 E4 r f# g=lastG trip{b-8 a g} c4~ c")
+        >>> fig_id1 = s.show('osmd')
+
+        To update a previously displayed score use the returned divId from before:
+
+        >>> fig_id2 = s.show('osmd', divId=fig_id1)
+        >>> assert fig_id1 == fig_id2
+        '''
+        in_ipython = runningUnderIPython()
+        score = obj
+        if fixPartName:
+            self.addDefaultPartName(score)
+
+        if divId is None:
+            # create unique reference to output div in case we wish to update it
+            divId = self.getUniqueDivId()
+
+
+        xml = open(score.write('musicxml')).read()
+        script = self.musicXMLToScript(xml, divId, offline=offline)
+        if in_ipython:
+            self.display(self.HTML(f'<div id="{divId}">loading OpenSheetMusicDisplay</div>'))
+            self.display(self.Javascript(script))
+        else:
+
+            tmp = tempfile.NamedTemporaryFile(delete=False)
+            tmp_path = tmp.name + '.html'
+
+            open(tmp_path, 'w').write(f"""
+            <div id="{divId}"></div>
+            <script>{script}</script>
+            """)
+            filename = 'file:///' + tmp_path
+            webbrowser.open_new_tab(filename)
+        return divId
+
+    def parseData(self):
+        raise NotImplementedError('osmd is display-only')
```

Comment:
This line is very easy to either exempt from coverage altogether with a `# pragma: no cover` after the signature, or else just actually test with a `self.assertRaises(NotImplementedError, myScore.show('osmd'))`

Reply:
used `#pragma: no cover` since seems neater, thanks.